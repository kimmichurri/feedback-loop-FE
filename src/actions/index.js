export const hasError = (message) => ({
  type: 'HAS_ERROR',
  message
})

export const isLoading = (boolean) => ({
  type: 'IS_LOADING',
  isLoading: boolean
})

export const setSurvey = (survey) => ({
  type: 'SET_SURVEY',
  survey
})

export const setCohorts = (cohorts) => ({
  type: 'SET_COHORTS',
  cohorts
})

export const setUser = (user) => ({
  type: 'SET_USER',
  user
})

export const setCurrentCohort = (cohort) => ({
  type: 'SET_CURRENT_COHORT',
  cohort
})