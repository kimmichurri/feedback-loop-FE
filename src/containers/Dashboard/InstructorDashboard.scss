.dashboard-container {
  background: $white;
  height: 500px;
  width: 80%;
  margin: auto;
  padding: 25px;
  border-radius: 5px;
  border-top: .65rem solid #ef3852;
  display: flex;
  flex-direction: column;
}

.inst-surveys {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.create-new-survey-button {
  background-color: $main-blue;
  border-radius: 50px;
  border: none;
  color: $accent-teal;
  font-family: $main-font;
  font-weight: bold;
  font-size: 16px;
  height: 50px;
  width: 80%;
  &:hover {
    cursor: pointer;
    background-color: $main-blue-hover;
  }
}
