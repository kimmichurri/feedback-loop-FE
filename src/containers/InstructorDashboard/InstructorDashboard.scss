.dashboard-container {
  background: $white;
  height: 500px;
  width: 80%;
  margin: auto;
  padding: 25px;
  border-top: .65rem solid #ef3852;
  box-shadow: 0 15px 20px rgba(0, 0, 0, 0.3);
  display: flex;
  flex-direction: column;
}

.inst-surveys {
  display: flex;
  flex-direction: column;
  align-items: center;
  overflow-y: scroll;
  height: 28em;
}

.create-new-survey-button {
  background-color: $main-blue;
  border-radius: 50px;
  margin-top: 2%;
  margin-left: 66%;
  box-shadow: 0 0 2px 1px rgba(0, 0, 0, 0.2);
  border: none;
  color: $accent-teal;
  font-family: $main-font;
  font-weight: bold;
  font-size: 16px;
  height: 50px;
  width: 200px;
  margin-bottom: 1%;
  &:hover {
    cursor: pointer;
    background-color: $main-blue-hover;
  }
}

.surveys-accordion {
  display: flex;
  flex-direction: column;
  justify-content: center;
}
