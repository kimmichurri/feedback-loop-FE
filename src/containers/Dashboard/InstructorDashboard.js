// import React, { Component } from 'react'
// import { NavLink } from 'react-router-dom'
//
// export class Dashboard extends Component {
//   render() {
//     return(
//       <div className='dashboard-container'>
//         <NavLink to='/new-survey'>
//           <button className='create-new-survey-button'>Create New Survey</button>
//         </NavLink>
//       </div>
//     )
//   }
// }
//
// export default Dashboard

import React, { Component } from 'react'
import { connect } from 'react-redux'
import { handleGet } from '../../thunks/handleGet'
import StudentSurvey from '../StudentSurvey/StudentSurvey'
import { setStudentSurveys } from '../../actions/'

export class InstructorDashboard extends Component {
  constructor() {
    super();
    this.state = {
      fakeSurveys: [
        { survey_name: 'Week 4 Survey',
          id: 4,
          questions: [
            { id: 1,
              questionTitle: 'How are you today?',
              options: [ { pointValue: 1, description: 'bad' }, { pointValue: 2, description: 'fine'}, { pointValue: 3, description: 'good' }, { pointValue: 4, description: 'great' } ]
            },
            { id: 2,
              questionTitle: 'What day is it?',
              options: [ { pointValue: 1, description: 'Thursday' }, { pointValue: 2, description: 'Friday'}, { pointValue: 3, description: 'Saturday' }, { pointValue: 4, description: 'Sunday' } ]
            }
          ]
        }
      ]
    }
  }

  async componentDidMount() {
    const url = `https://turing-feedback-api.herokuapp.com/api/v1/surveys/${this.props.user.api_key}`
    // const surveys = await this.props.handleGet(url)
    // // const surveys = this.state.fakeSurveys
    // this.setState({
    //   surveys: surveys
    // })
    const surveys = this.state.fakeSurveys
    this.props.setStudentSurveys(surveys)
  }

  listSurveys = () => {
    const surveys = this.state.fakeSurveys
    return surveys.map(survey => {
      return <button key={survey.id} onClick={() => {this.renderSurvey(survey)}}>{survey.survey_name}</button>
    })
  }

  renderSurvey = (survey) => {
    this.props.history.push(`/student-survey/${survey.survey_name}`)
    return <StudentSurvey />
  }

  render() {
    return(
      <div className='dashboard-container'>
        Student Dashboard
        {this.listSurveys()}
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export const mapDispatchToProps = (dispatch) => ({
  handleGet: (url) => dispatch(handleGet(url)),
  setStudentSurveys: (surveys) => dispatch(setStudentSurveys(surveys))
})

export default connect(mapStateToProps, mapDispatchToProps)(InstructorDashboard)
