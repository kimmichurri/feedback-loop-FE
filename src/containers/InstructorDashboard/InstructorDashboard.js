import React, { Component } from 'react'
import { connect } from 'react-redux'
import { handleGet } from '../../thunks/handleGet'
import { Link } from "react-router-dom";
import SurveyCard from '../SurveyCard/SurveyCard'
import { setInstructorSurveys } from '../../actions'
import PropTypes from 'prop-types'

export class InstructorDashboard extends Component {

  async componentDidMount() {
    const myKey = await localStorage.getItem('currentUser')
    const url = `https://turing-feedback-api.herokuapp.com/api/v1/surveys?api_key=${myKey}`
    const surveys = await this.props.handleGet(url)
    this.props.setInstructorSurveys(surveys)
  }

  render() {
    return(
      <div className='surveys-accordion'>
        <h2 className='inst-dashboard-header'>My Dashboard</h2>
        <div className='inst-surveys'>
          {this.props.instructorSurveys && this.props.instructorSurveys.map(survey => {
            return <SurveyCard key={survey.id}
                               surveyData={survey}/>
          })}
        </div>
        <Link to='/new-survey'>
          <button className='create-new-survey-button'>Create New Survey</button>
        </Link>
      </div>
    )
  }
}

InstructorDashboard.propTypes = {
  user: PropTypes.string,
  instructorSurveys: PropTypes.array,
  handleGet: PropTypes.func,
  setInstructorSurveys: PropTypes.func
}

export const mapStateToProps = (state) => ({
  user: state.user,
  instructorSurveys: state.instructorSurveys
})

export const mapDispatchToProps = (dispatch) => ({
  handleGet: (url) => dispatch(handleGet(url)),
  setInstructorSurveys: (surveys) => dispatch(setInstructorSurveys(surveys))
})

export default connect(mapStateToProps, mapDispatchToProps)(InstructorDashboard)