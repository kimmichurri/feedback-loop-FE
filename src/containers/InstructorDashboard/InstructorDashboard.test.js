import React from 'react'
import { shallow } from 'enzyme'
import { InstructorDashboard , mapStateToProps, mapDispatchToProps} from './InstructorDashboard'
import { setInstructorSurveys } from '../../actions'
jest.mock('../../thunks/handleGet')

describe('InstructorDashboard', () => {
  let wrapper
  let handleGet
  let user
  let mockSetInstructorSurveys
  let mockInstructorSurveys
  let mockDispatch

  beforeEach(() => {
    mockInstructorSurveys = [
      { id: 1, name: "Survey 1"},
      { id: 2, name: "Survey 2"}
    ]
    handleGet = jest.fn()
    mockSetInstructorSurveys = jest.fn()
    user = "2nf9rnad"
    mockDispatch = jest.fn()
    wrapper = shallow(
      <InstructorDashboard
        handleGet={handleGet}
        user={user}
        instructorSurveys={mockInstructorSurveys}
        setInstructorSurveys={mockSetInstructorSurveys}
      />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()  
  })

  describe('mapStateToProps', () => {
    it('should return a user and instructor surveys as props', () => {
      const mockState = {
        user: "2nf9rnad",
        instructorSurveys: [
          { id: 1, name: "Survey1"},
        { id: 2, name: "Survey2"}
        ],
        fakeState: "fakeState"
      }
      const expected = {
        user: "2nf9rnad",
        instructorSurveys: [
          { id: 1, name: "Survey1"},
        { id: 2, name: "Survey2"}
        ]
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should return handleGet as props', () => {
      const mockUrl = 'www.getSurveys.com'
      const handleGet = jest.fn()
      const actionToDispatch = handleGet(mockUrl)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.handleGet(mockUrl)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should return setInstructorSurveys as props', () => {
      const actionToDispatch = setInstructorSurveys(mockInstructorSurveys)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setInstructorSurveys(mockInstructorSurveys)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })

})