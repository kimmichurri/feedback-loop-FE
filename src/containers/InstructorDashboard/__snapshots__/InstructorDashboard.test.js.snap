// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructorDashboard should match the snapshot 1`] = `
<div
  className="surveys-accordion"
>
  <Link
    to="/new-survey"
  >
    <button
      className="create-new-survey-button"
    >
      Create New Survey
    </button>
  </Link>
  <div
    className="inst-surveys"
  >
    <SurveyCard
      key="1"
      surveyData={
        Object {
          "id": 1,
          "name": "Survey 1",
        }
      }
    />
    <SurveyCard
      key="2"
      surveyData={
        Object {
          "id": 2,
          "name": "Survey 2",
        }
      }
    />
  </div>
</div>
`;
